### DISCUSSION TITLE
Read instead of watch?

### DISCUSSION THREAD

**Discussion by busslina** [GregoryConrad/rearch-dart#74] [Q&A]
I want to use an analogous to Riverpod's read: I want to read the value of a capsule inside a `callonce` side effect. Is it possible? Or I must `use` the capsule?

I want to prevent rebuilds when the readed capsule updates, because I'm inside a `callonce` side effect.

**Comment by GregoryConrad** [comment]
What are you doing in the `callonce` side effect that requires the `use`? That seems a little strange, as it breaks reactivity.

**Reply by busslina** [reply]
Again, migrating old code...

```dart
use.callonce(() {
          serversScreenController.setRequestFocus((previous) {
            if (!connection.state.isConnected ||
                !dynamicTabsController.requestFocus()) {
              previous();
            }
          });
        });
```

Finally, I use the capsule before `callonce` because I need `connection` in other part... but seems that in some case will be desired a non reactive `read` way.

Here I delegate the focus requesting to the parent either if it isn't connected or if the tab focus request was denied. This design worked really well with the former design (Riverpod). Maybe I want to re-design it. But I think that `read` could be needed sometimes.

**Reply by GregoryConrad** [reply]
Hmm, I think I'd need a full example to figure out the "reactive" way to do what you are trying to do. For migration purposes, it is fine to `use()` in the `callonce` and eat the (possible) widget rebuild because it shouldn't be that much of a performance hit.

**Reply by busslina** [reply]
Anyway, I just started today a fresh lib+project (for testing PostgreSQL) so I will explore all the benefits of ReArch in deep. (I needed this, too many days migrating was killing me...)
